src/account/components/OptionalAuthTopMessage.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+import React from 'react';
+
+import _ from 'lodash/fp';
+import { Trans } from 'react-i18next';
+import { useSelector } from 'react-redux';
+
+import { Typography } from '@mui/material';
+
+import RouterLink from 'common/components/RouterLink';
+import { useOptionalAuth } from 'navigation/components/Routes';
+
+const OptionalAuthTopMessage = () => {
```

Comment:
Question: this looks to be substantially the same as `OptionalAuthBottomMessage`. Can they be combined?

Reply:
@paulschreiber I think potentially we could do that, but I dont know if it will be better since the structure is not the same, the styles are not the same and the messages are not the same, so all of that should be configured dynamically with props that IMO will make this more difficult to understand

Reply:
What are these messages doing?